utils/ifupdown.sh.in

```diff
@@ -93,15 +93,36 @@ done
 # We finish setting up the bridge
 if [ "$MODE" = 'start' ]; then
 
+  # check if mstpd is running before bridge-stp may spawn it
+  found=$(pidof -c -s mstpd)
+
   # This triggers the kernel to run `/sbin/bridge-stp start $IFACE`
-  "$brctl" stp "$IFACE" on
+  "$ip" link set "$IFACE" type bridge stp_state 1
+
+  # if mstpd is just spawned, let it initialize
+  if [ -z "$found" ]; then
+    maxwait=2
+    # Use 0.1 delay if available
+    sleep 0.1 2>/dev/null && maxwait=$((maxwait*10))
+    count=0
+    while [ -z "$found" ] && [ $count -lt $maxwait ]; do
+      found=$(pidof -c -s mstpd)
```

Comment:
I must be missing something ... Why is this necessary?  I'm pretty sure `brctl stp <iface> on` didn't return until after `/sbin/bridge-stp` returned, and `mstpd` was guaranteed to be running when `/sbin/bridge-stp` returned.  Does `ip link ... stp_state 1` behave differently and return early?

Reply:
They behave the same (both wait), but there is a race which I can reproduce on both brctl and ip  link... just run:
```
# brctl stp <bridge> on; mstpctl addbridge <bridge>
ctl_client_init: Couldn't connect to server
can't setup control connection

```
Even the slightest delay (sleep 0.1) seems to overcome it, so I'm guessing mstpd backgrounds before it's ready to receive connections... this small loop is easy enough though :)

Reply:
Hmm... Ok.  In that case, I think this loop/delay should probably go in the bridge-stp script (after spawning mstpd) instead of in ifupdown.sh.  bridge-stp could be used on systems that don't use ifupdown.sh (that use a different script for configuration), so let's solve this for everyone instead of only for those systems that use ifupdown.sh.

Reply:
OK, wasn't sure that would work given the warnings about kernel locks in the bridge-stp.in script, but I tried it and a brief sleep there allowed the client process to connect to mstpd, so I'll do that and revert the ifupdown loop.